.pre-commit-config.yaml

```diff
@@ -30,6 +30,10 @@ repos:
         files: ^pandas
         exclude: ^pandas/tests
         args: [--select, "ANN001,ANN2", --fix-only, --exit-non-zero-on-fix]
+    -   id: ruff
+        args: [--exit-non-zero-on-fix, --select=TID251]
```

Comment:
Cant we just add `TID251` to the `pyproject.toml`?

Reply:
@mroeschke indeed all the `TID` rules are already enabled. This new hook is redundant because the first `ruff` hook can cover it already.

Do you think I need to ignore `TID251` in `pandas/tests/*`? If not, this PR should be ready for review

Reply:
`TID251` shouldn't be ignored for tests